modules/mono/editor/bindings_generator.cpp

```diff
@@ -2789,6 +2797,35 @@ Error BindingsGenerator::_generate_cs_method(const BindingsGenerator::TypeInterf
 						"' from the editor API. Core API cannot have dependencies on the editor API.");
 	}
 
+	if (p_imethod.is_virtual && p_use_span) {
+		return OK;
+	}
+
+	bool has_span_argument = false;
+
+	if (p_use_span) {
+		for (const ArgumentInterface &iarg : p_imethod.arguments) {
+			const TypeInterface *arg_type = _get_type_or_singleton_or_null(iarg.type);
+			ERR_FAIL_NULL_V_MSG(arg_type, ERR_BUG, "Argument type '" + iarg.type.cname + "' was not found.");
+
+			if (arg_type->is_span_compatible) {
+				has_span_argument = true;
+				break;
+			}
+		}
+
+		if (p_imethod.is_vararg) {
+			has_span_argument = true;
+		}
+	}
+
+	if (p_use_span && has_span_argument) {
+		// Span overloads use the same method bind as the array overloads.
+		// Since both overloads are generated one after the other, we can decrease the count here
+		// to ensure the span overload uses the same method bind.
+		p_method_bind_count--;
```

Comment:
```suggestion

		if (has_span_argument) {
			// Span overloads use the same method bind as the array overloads.
			// Since both overloads are generated one after the other, we can decrease the count here
			// to ensure the span overload uses the same method bind.
			p_method_bind_count--;
		}
```
Makes it clearer how things relate as it can't be valid by any other path